File: JavaPrototypes/ExpenseCalculator/src/com/example/expenseCalculator/Manager.java
package com.example.expenseCalculator;

import java.util.ArrayList;
import java.util.List;

public class Manager implements Employee, Organizational {

	private String title = "";
	private Double expenseAllocation = (double) 0;

	public String getTitle() {
		return title;
	}

	public void setTitle(String value) {
		title = value;
	}

	public double getExpenseAllocation() {
		return expenseAllocation;
	}

	public void setExpenseAllocation(double value) {
		expenseAllocation = value;
	}
	
	public Manager(String title, double expenseAllocation)
	{
		this.title = title;
		this.expenseAllocation = expenseAllocation;
	}
	
	private List<Employee> employees = new ArrayList<Employee>();

	public List<Employee> getEmployees() {
		return employees;
	}

	public double getTotalExpenses() {
		double expenses = 0;
		
		for(Employee employee : employees)
		{
			if (employee.getClass() ==  Manager.class)
				expenses += ((Manager)employee).getTotalExpenses();
			else
				expenses += employee.getExpenseAllocation();
		}
		
		return expenses + this.getExpenseAllocation();
	}
}

File: DotNetPrototypes/ExpenseCalculator/Manager.cs
using System.Collections.Generic;

namespace ExpenseCalculator 
{
    public class Manager : IEmployee, IOrganizational
    {
        #region IEmployee Members

        public string Title
        {
            get;
            set;
        }

        public decimal ExpenseAllocation
        {
            get;
            set;
        }

        #endregion

        public Manager(string title, decimal expenseAllocation)
        {
            this.Title = title;
            this.ExpenseAllocation = expenseAllocation;
        }

        #region IOrganizational Members

        private List<IEmployee> employees = new List<IEmployee>();

        public List<IEmployee> Employees
        {
            get { return employees; }
        }

        public decimal GetTotalExpenses()
        {
            decimal expenses = 0;
            foreach (IEmployee employee in employees)
            {
                if (employee.GetType() == typeof(Manager))
                    expenses += ((Manager)employee).GetTotalExpenses();
                else
                    expenses += employee.ExpenseAllocation;
            }

            return expenses + ExpenseAllocation;
        }

        #endregion
    }
}
